packages/file-types/zarr/src/anndata-loaders/ObsFeatureMatrixAnndataLoader.js

```diff
@@ -10,6 +10,18 @@ import {
 // value returned from the load function.
 const toObject = data => ({ data });
 
+const filterOutEveryOther = (arr) => {
+  const filtered = new arr.constructor(arr.length / 2);
+  for (let i = 0; i < (arr.length / 2); i++) {
+    filtered[i] = arr[i * 2];
+  }
+  return filtered;
+};
+// eslint-disable-next-line no-undef
+const maybeDowncastInt64 = data => (data.constructor === BigInt64Array
+  ? filterOutEveryOther(new Int32Array(data.buffer))
+  : data);
+
```

Comment:
@manzt Are you aware of a faster or better way of going from BigInt64Array to Int32Array? Some comparisons and alternatives:

```javascript
const filterOutEveryOther = (arr) => {
  const filtered = new arr.constructor(arr.length / 2);
  for (let i = 0; i < (arr.length / 2); i++) {
    filtered[i] = arr[i * 2];
  }
  return filtered;
};

const castByBuffer = (arr) => {
  const buffer = new ArrayBuffer(arr.length * 4)
  const filtered = new DataView(buffer);
  const viewOfArr = new DataView(arr.buffer)
  for (let i = 0; i < arr.length; i++) {
    filtered.setInt32(i * 4, viewOfArr.getInt32(i * 8, false));
  }
  return new Int32Array(buffer);
};

const castByArr = (arr) => {
  const filtered = new Int32Array(arr.length);
  for (let i = 0; i < arr.length; i++) {
    filtered[i] = Number(arr[i])
  }
  return filtered;
};


let data = new BigInt64Array([1n, 3n, 10n, 15n])
console.log('ops produce equal results?', castByArr(data), castByBuffer(data), filterOutEveryOther(new Int32Array(data.buffer)))
data = BigInt64Array.from(Array.from(Array(10000000).keys()).map(BigInt))
let t = performance.now()
filterOutEveryOther(new Int32Array(data.buffer))
console.log('filtering took ', performance.now() - t)

t = performance.now()
castByBuffer(data)
console.log('cast by buffer took ', performance.now() - t)

t = performance.now()
castByArr(data)
console.log('cast by arr took ', performance.now() - t)
```